### ISSUE TITLE
A user failed to initialize (0x8000FFFF), setting folder permissions is invalid, and a timeout occurs when running with administrator permissions (0x800705B4)

### ISSUE THREAD

**Issue by sln162** [MicrosoftEdge/WebView2Feedback#3149]
A user reported this problem, running the test program with non-administrator privileges, reporting an error during initialization: (0x8000FFFF): Catastrophic failure (Exception from HRESULT: 0x8000FFFF (E_UNEXPECTED), as shown in the screenshot.
The software location will generate a UDF folder, and setting the UDF folder permissions to add the full control permissions of Users, Administrators, and Everyone is still the same invalid. Run the software with administrator privileges, initialization will time out after 130 seconds(0x800705B4).

I have collected ETW logs of the initialization process.It may be a customer system failure, please analyze it according to the ETW file.

<img width="300" alt="未命名1674203788" src="https://user-images.githubusercontent.com/11196986/213841474-0e427bfe-89fa-4cf1-82eb-e3711315f306.png">

Test program source code:
[WebView2Feedback.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/10412238/WebView2Feedback.zip)
ETW file:
[trace0120.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/10471402/trace0120.zip)
UDF folder:
[WebView2Feedback.exe.WebView2.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/10471426/WebView2Feedback.exe.WebView2.zip)
![image](https://user-images.githubusercontent.com/11196986/213841899-6d545bc7-ea1e-4601-aa70-a3315239b038.png)

Version
SDK: 1.0.1518.46
Runtime:   107 and  109.0.1518.61
Framework: .Net 4.5.2
OS: Win10 19045

[AB#43102576](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/43102576)

**Comment by LiangTheDev** [comment]
From the ETW trace, it suggests that msedgewebview2.exe crashpad process crashed shortly after it started. It should be that the "full control permissions" have not been set to sub folders and files that have already been created.

The best way to figure out would be to take a procmon trace and looking for Access Denied file accesses from msedgewebview2.exe.

**Comment by sln162** [comment]
@LiangTheDev  At that time, the "full control permission" was set to be attached to the subfolder, which seemed normal. It is estimated that it is a problem with his system. Next time I encounter it, I will try procmon to track it.

Regarding the permission issue, only UDF needs to care about the permission, and does the WebView2 installation folder need to consider the permission issue?

**Comment by LiangTheDev** [comment]
Only UDF needs to care about permission. WebView2 installation folder should allow all applications/webview processes access, unless the user or software on the device messes up with default permissions for related folders.

**Comment by sln162** [comment]
@LiangTheDev     
> unless the user or software on the device messes up with default permissions for related folders.

Then this may happen, should the permission check of the installation directory be increased?
We occasionally see the following errors in the background:
The system can not find the file specified. (Exception from HRESULT:0x80070002)
access denied. (Exception from HRESULT:0x80070005 (E_ACCESSDENIED))

Are these two initialization errors caused by insufficient permissions?

**Comment by LiangTheDev** [comment]
File not found is more likely due to Edge WebView2 Runtime not installed or corrupted. Install info (https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/install.md) would be helpful, and a procmon log could provide even more info. 

Access denied could comes from multiple places. We've encountered access deny due to UDF permission, but have not see one due to installation folder. A procmon log is needed to know what's really going on.

**Comment by sln162** [comment]
@LiangTheDev    Thanks for the reply, I have added the "system cannot find the file" processing method, will locate the installation folder through the registry, and check whether "msedgewebview2.exe" is really missing, if so, trigger the function of reinstalling WebView2.

Access denied. I saw it once when I set the permission of "msedgewebview2.exe" to deny the read permission of "Users". The user who actually encountered this problem did not contact me. If a user contacts me later, I will Start a new question.
